csm/conf/post_install.py

```diff
@@ -127,10 +128,26 @@ def validate_3rd_party_pkgs(self):
             raise CsmSetupError(f"Failed at package Validation: {ve}")
 
     def set_ssl_certificate(self):
-        ssl_certificate = Conf.get(const.CONSUMER_INDEX, self.conf_store_keys[const.KEY_SSL_CERTIFICATE])
-        Conf.set(const.CSM_GLOBAL_INDEX, const.SSL_CERTIFICATE_PATH, ssl_certificate)
-        Conf.set(const.CSM_GLOBAL_INDEX, const.PRIVATE_KEY_PATH_CONF, ssl_certificate)
-        Log.info(f"Setting ssl certificate path: {ssl_certificate}")
+        ssl_certificate_path = Conf.get(const.CONSUMER_INDEX, self.conf_store_keys[const.KEY_SSL_CERTIFICATE])
+        csm_endpoint = Conf.get(const.CONSUMER_INDEX, const.CSM_AGENT_ENDPOINTS_KEY)
+        csm_protocol, csm_host, csm_port = self._parse_endpoints(csm_endpoint)
+        if csm_protocol == 'https' and not os.path.exists(ssl_certificate_path):
+            Log.warn(f"HTTPS enabled but SSL certificate not found at: {ssl_certificate_path}")
+            Log.info(f"Generating self signed ssl certificate")
```

Comment:
"Generating self signed ssl certificate" should be combined with log.warn

Reply:
done